redis/commands/core.py

```diff
@@ -369,16 +369,14 @@ class ManagementCommands(CommandsProtocol):
     Redis management commands
     """
 
-    def auth(self, password, username=None, **kwargs):
+    def auth(self, password, username="default", **kwargs):
         """
         Authenticates the user. If you do not pass username, Redis will try to
         authenticate for the "default" user. If you do pass username, it will
         authenticate for the given user.
         For more information see https://redis.io/commands/auth
         """
-        if username:
-            return self.execute_command("AUTH", username, password, **kwargs)
-        return self.execute_command
+        return self.execute_command("AUTH", username, password, **kwargs)
```

Comment:
Given the [command docs](https://redis.io/commands/auth/) it would appear instead that the following is more accurate. Thoughts?
```
def auth(self, password, username=None, **kwargs):
    parts = []
    if username is not None:
        parts.append(username)
    parts.append(password)
    return self.execute_command("AUTH", *parts, **kwargs)
```

Reply:
Thanks for the suggestion, I like it more. But there's a problem, if no password is set for the default user, `AUTH` command will fail with the following error. Won't that be a problem? 
`(error) ERR AUTH <password> called without any password configured for the default user. Are you sure your configuration is correct?`

However, `AUTH default ""` returns OK.